### PR TITLE
Cleanup native docker containers

### PR DESCRIPTION
**Pull request by cjellick**

### CODE REVIEW

File: tests/validation/cattlevalidationtest/core/test_native_docker.py [hunk 1]
```diff
@@ -59,164 +59,193 @@ def pull_images(docker_client):
...
-    r_containers = admin_client.list_container(name=name)
-    assert len(r_containers) == 1
-    c = r_containers[0]
-    c = admin_client.wait_success(c)
-
-    assert c.externalId == d_container['Id']
-    assert c.state == 'running'
-    assert c.primaryIpAddress == inspect['NetworkSettings']['IPAddress']
+def test_native_unmanaged_network(docker_client, admin_client, pull_images,
+                                  request):
```
Comment (sonchang) [inline tests/validation/cattlevalidationtest/core/test_native_docker.py]: You added a request object.  Were you intending to add a finalizer?

**Comment by alena1108** [comment]
LGTM

File: tests/validation/cattlevalidationtest/core/test_native_docker.py [hunk 2]
```diff
@@ -59,164 +59,193 @@ def pull_images(docker_client):
...
 
-    nics = super_client.reload(c).nics()
-    assert len(nics) == 1
-    assert c.primaryIpAddress == nics.data[0].ipAddresses().data[0].address
+        nics = super_client.reload(container).nics()
+        assert len(nics) == 1
+        assert container.primaryIpAddress == nics.data[0].ipAddresses().data[
+            0].address
+    finally:
+        cleanup_container(admin_client, container)
```
Comment (sonchang) [inline tests/validation/cattlevalidationtest/core/test_native_docker.py]: quick question: in the other 2 tests, we declare the container to be None outside of the try block.  i'm still a python newbie.  do we need to do that here too?

**Comment by sonchang** [comment]
LGTM

**Pull request merged**